check/src/cache.rs

```diff
@@ -34,89 +38,218 @@ sol! {
+        }
     }
-    let cache_manager = *cache_manager_addrs.last().unwrap();
-    let cache_manager = H160::from_slice(cache_manager.as_slice());
+}
 
-    let contract: Address = cfg.address.to_fixed_bytes().into();
-    let data = CacheManager::placeBidCall { program: contract }.abi_encode();
-    let mut tx = Eip1559TransactionRequest::new()
-        .to(cache_manager)
-        .data(data);
+/// Checks the status of the Stylus cache manager, including the cache size, queue size, and minimum bid
+/// for different contract sizes as reference points. It also checks if a specified Stylus contract address
+/// is currently cached.
+pub async fn check_status(cfg: &CacheStatusConfig) -> Result<()> {
+    let provider = ProviderBuilder::new()
+        .with_recommended_fillers()
+        .on_builtin(&cfg.endpoint)
+        .await?;
+    let arb_wasm_cache = ArbWasmCache::new(ARB_WASM_CACHE_ADDRESS, provider.clone());
+    let cache_manager_addr = get_cache_manager_address(provider.clone()).await?;
+    let cache_manager = CacheManager::new(cache_manager_addr, provider.clone());
+    let CacheManager::isPausedReturn { _0: is_paused } = cache_manager.isPaused().call().await?;
+    let CacheManager::queueSizeReturn { _0: queue_size } = cache_manager.queueSize().call().await?;
+    let CacheManager::cacheSizeReturn { _0: cache_size } = cache_manager.cacheSize().call().await?;
+    let CacheManager::getMinBid_1Return { min: min_bid_smol } = cache_manager
+        .getMinBid_1(ByteSize::kb(8).as_u64())
+        .call()
+        .await?;
+    let CacheManager::getMinBid_1Return { min: min_bid_med } = cache_manager
+        .getMinBid_1(ByteSize::kb(16).as_u64())
+        .call()
+        .await?;
+    let CacheManager::getMinBid_1Return { min: min_bid_big } = cache_manager
+        .getMinBid_1(ByteSize::kb(24).as_u64())
+        .call()
+        .await?;
 
-    // If a bid is set, specify it. Otherwise, a zero bid will be sent.
-    if let Some(bid) = cfg.bid {
-        tx = tx.value(U256::from(bid));
-        greyln!("Setting bid value of {} wei", bid.debug_mint());
+    greyln!(
+        "Cache manager address: {}",
+        cache_manager_addr.debug_lavender()
+    );
+    greyln!(
+        "Cache manager status: {}",
+        if is_paused {
+            "paused".debug_red()
+        } else {
+            "active".debug_mint()
+        }
+    );
+    let cache_size = ByteSize::b(cache_size);
+    let queue_size = ByteSize::b(queue_size);
+    greyln!("Cache size: {}", cache_size.debug_grey());
+    greyln!("Queue size: {}", queue_size.debug_grey());
+    greyln!(
+        "Minimum bid for {} contract: {}",
+        "8kb".debug_mint(),
+        min_bid_smol.debug_lavender()
+    );
+    greyln!(
+        "Minimum bid for {} contract: {}",
+        "16kb".debug_yellow(),
+        min_bid_med.debug_lavender()
+    );
+    greyln!(
+        "Minimum bid for {} contract: {}",
+        "24kb".debug_red(),
+        min_bid_big.debug_lavender()
+    );
```

Comment:
```suggestion
    greyln!(
        "Minimum bid for {} {} {}",
        "8kb".debug_mint(),
        "contract:".grey(),
        min_bid_smol.debug_lavender()
    );
    greyln!(
        "Minimum bid for {} {} {}",
        "16kb".debug_yellow(),
        "contract:".grey(),
        min_bid_med.debug_lavender()
    );
    greyln!(
        "Minimum bid for {} {} {}",
        "24kb".debug_red(),
        "contract:".grey(),
        min_bid_big.debug_lavender()
    );
```
Contract shows as white instead of grey.